docs/design/Path concatenation.md

```diff
@@ -0,0 +1,94 @@
+## Paths in DF2
+
+It is expected that the use of paths in DF2 (similar in nature to Unix file paths, but not exactly the same) will be extensive, as will be concatention operations on them: `GetFullPath` and `GetRelativePath`. This document will provide a specification for how these operations are to behave as well as contain some considerations that will drive the design of concatenation functions.
+
+## Definitions
+
+_Char_ is a Unicode code point. There are special _chars_ which have specific meaning in DF2 paths:
+- _Separator_ is a _char_ corresponding to the code point `U+002F` - `Slash`.
+- _Dot_ is a _char_ corresponding to the code point `U+002E` - `Full stop`.
+
+Two _chars_ are equal if they refer to the same code point.
+
+[Rationale: to support the performance of concatenation functions, it is neccessary to impose the restriction that equality comparisons be strictly ordinal.]
+
+_Segments_ are sequences of _chars_ that have no _separators_ in them. There are special _segments_ which have specific meaning in DF2 paths:
+- _Skip segment_ is a _segment_ that consists of one _dot_ or has no _chars_ at all.
+- _Backtrack segment_ is a _segment_ that consists of two _dots_.
+
+Two _segments_ are equal if all their _chars_ are equal, except for _skip segments_, which are always equal to each other.
+
+DF2 _paths_ are sequences of _segments_ and _separators_. It is illegal for two _segments_ to follow each other. Two _separators_ that have no _chars_ between them are treated as a _separator_-_skip segment_-_separator_ triplet.
```

Comment:
Why is it illegal for two segments to follow each other?
`a` is a valid segment and `b` is also a valid segment.
`ab` should be perfectly valid. I get what this means to say, but I think it's more confusing then helpful.

Reply:
> Why is it illegal for two segments to follow each other?

So that it is unambiguous that  e. g. `path` is not 4 segments or an infinite number of empty segments, etc.

> I get what this means to say, but I think it's more confusing then helpful.

I will think of a better way to express this invariant.

Reply:
```suggestion
DF2 _paths_ are sequences of _segments_ and _separators_. Two _separators_ that have no _chars_ between them are treated as a _separator_-_skip segment_-_separator_ triplet. It is illegal for two _skip segments_ to follow each other. 
```

Maybe? (If i'm understanding your comment correctly)

Reply:
This would still leave it ambiguous whether `path` is `path` or `pa` + `th` unfortunately.
I am thinking of something like:
```
...a segment can only be followed by a separator...
```
This would eliminate the ambiguity while being a little clearer on the intent. @HurricanKai thoughts?

Reply:
Sounds good to me, but I think it can either be followed by a separator, or nothing.